src/server/string_family.cc

```diff
@@ -525,11 +525,41 @@ SinkReplyBuilder::MGetResponse OpMGet(bool fetch_mcflag, bool fetch_mcver, const
   response.storage_list = SinkReplyBuilder::AllocMGetStorage(total_size);
   char* next = response.storage_list->data;
 
+  auto log_error = [](auto it, auto expected_key, auto key_found) {
+    LOG(ERROR) << "Invalid type " << it->second.ObjType();
+    LOG(ERROR) << "Expected key " << expected_key;
+    LOG(ERROR) << "Key found " << key_found;
+    LOG(ERROR) << "Bucket id: " << it.bucket_id();
+    LOG(ERROR) << "Segment " << it.segment_id();
+    LOG(ERROR) << "Slot id " << it.slot_id();
+  };
+
   for (size_t i = 0; i < args.size(); ++i) {
     PrimeConstIterator it = iters[i];
     if (it.is_done())
       continue;
 
+    std::string key_found;
+    it->first.GetString(&key_found);
+
+    log_error(it, args[i], key_found);
```

Comment:
Unconditional log

Reply:
I was not clear. I would like to log everything (all the iterators) only if there is a mismatch and we enter line 548.